src/api/l_math_vectors.c

```diff
@@ -857,9 +942,9 @@ static int l_lovrVec4Add(lua_State* L) {
   if (lua_type(L, 2) == LUA_TNUMBER) {
     float x = lua_tonumber(L, 2);
     v[0] += x;
-    v[1] += x;
-    v[2] += x;
-    v[3] += x;
+    v[1] += luax_optfloat(L, 3, x);
+    v[2] += luax_optfloat(L, 4, x);
+    v[3] += luax_optfloat(L, 5, x);
```

Comment:
I wondered if it would be faster to do all the optfloats first, then do the math, but it didn't appear to make a difference.